crates/stages/src/stages/hashing_account.rs

```diff
@@ -33,6 +34,81 @@ impl Default for AccountHashingStage {
     }
 }
 
+#[derive(Clone, Debug)]
+/// `SeedOpts` provides configuration parameters for calling `AccountHashingStage::seed`
+/// in unit tests or benchmarks to generate an initial database state for running the
+/// stage.
+///
+/// In order to check the "full hashing" mode of the stage you want to generate more
+/// transitions than `AccountHashingStage.clean_threshold`. This requires:
+/// 1. Creating enough blocks + transactions so there's enough transactions to generate
+/// the required transition keys in the `BlockTransitionIndex` (which depends on the
+/// `TxTransitionIndex` internally)
+/// 2. Setting `transitions > clean_threshold` so that there's enough diffs to actually
+/// take the 2nd codepath
+pub struct SeedOpts {
+    /// The range of blocks to be generated
+    pub blocks: Range<u64>,
+    /// The range of accounts to be generated
+    pub accounts: Range<u64>,
+    /// The range of transactions to be generated per block.
+    pub txs: Range<u8>,
+    /// The number of transitions to go back, capped at the number of total txs
+    pub transitions: u64,
+}
+
+#[cfg(any(test, feature = "test-utils"))]
+impl AccountHashingStage {
+    /// Initializes the `PlainAccountState` table with `num_accounts` having some random state
+    /// at the target block, with `txs_range` transactions in each block.
+    ///
+    /// Proceeds to go to the `BlockTransitionIndex` end, go back `transitions` and change the
+    /// account state in the `AccountChangeSet` table.
+    pub fn seed<DB: Database>(
+        tx: &mut Transaction<'_, DB>,
+        opts: SeedOpts,
+    ) -> Result<Vec<(Address, Account)>, StageError> {
+        use reth_db::models::AccountBeforeTx;
+        use reth_interfaces::test_utils::generators::{
+            random_block_range, random_eoa_account_range,
+        };
+        use reth_primitives::{H256, U256};
+        use reth_provider::insert_canonical_block;
+
+        let blocks = random_block_range(opts.blocks, H256::zero(), opts.txs);
+        let num_transitions = blocks.iter().map(|b| b.body.len() as u64).sum();
+        let transitions = std::cmp::min(opts.transitions, num_transitions);
+
+        for block in blocks {
+            insert_canonical_block(&**tx, &block, true).unwrap();
+        }
+
+        // Account State generator
+        let accounts = random_eoa_account_range(opts.accounts);
```

Comment:
not for this PR, but we should unify the random generation of types into `arbitrary/proptest` instead of `rand`

Reply:
https://github.com/paradigmxyz/reth/issues/1306